### PR TITLE
Make Panel.Header use a h2

### PR DESCRIPTION
**Pull request by toolmantim**
In general it's good to use semantic tags for accessibility, but not for styling, so let's use a h2 here for panel headers.

### CODE REVIEW

File: app/components/shared/Panel/index.js [hunk 1]
```diff
@@ -39,9 +40,9 @@ Panel.IntroWithButton = IntroWithButton;
 Panel.Row = Row;
 Panel.RowActions = RowActions;
 Panel.RowLink = RowLink;
+Panel.Header = Header;
 
 const SIMPLE_COMPONENTS = {
-  Header: 'bg-silver py2 px3 semi-bold',
```
Comment (ticky) [inline app/components/shared/Panel/index.js]: I’d really like to make it so such element changes don’t require us to be splitting these back out into files like this. I was thinking about this today while I was doing #230 :(

Comment (toolmantim) [inline app/components/shared/Panel/index.js]: I figured it was inevitable due to the combination of linter rules not allowing multiple components in a file, and the general inflexibility of this type of code: 

```
const SIMPLE_COMPONENTS = {
  Description: 'dark-gray mt1',
  Icon: 'flex items-center',
  Menu: 'flex items-center'
};

Object.keys(SIMPLE_COMPONENTS).forEach((componentName) => {
```

Comment (toolmantim) [inline app/components/shared/Panel/index.js]: Just to clarifiy, I'm talking about how this code assumes a component only has two things: an identifier, and a list of class names.

Whereas if we just always used `React.PureComponent` for things, and made them declarative in JSX, then everything in the system is the same, and it's easy for anyone to, say, change a tag name, or a class name, or add an event handler, etc. And if we want to DRY some logic into a single place, we do that using our own functions calls from the components?

I just always find I regret defining classes/functions/things imperatively in loops like this, as it just makes it hard for anyone following me to fix/customize/adapt things.

Comment (toolmantim) [inline app/components/shared/Panel/index.js]: I'm going to merge this, but I'm keen to hear your thoughts still!

**Pull request merged**